### PR TITLE
Support mounting secrets and config maps.

### PR DESCRIPTION
**Pull request by igsong**
From Knative 0.4.0, secrets and config maps can be mounted.
To support the feature, I've added --mount-config-map and --mount-secret flags.
In addition, the version of knative/build and knative/serving are bumped up to 0.5.0 and label selector for getting istio ingress is changed to "istio:ingressgateway" instead of "knative:ingressgateway" to reflect changes introduced from Knative 0.4.0.

### CODE REVIEW

File: pkg/knctl/cmd/service/deploy.go [hunk 1]
```diff
@@ -109,6 +117,22 @@ func (o *DeployOptions) Run() error {
...
+		if err != nil {
+			return err
+		} 
+		
+		d, err := yaml.Marshal(&svc)
+		if err != nil {
+			return err
+		}
+
+		fmt.Printf("%s\n", d)
```
Comment (cppforlife) [inline pkg/knctl/cmd/service/deploy.go]: should be `o.ui.PrintBlock(d)`

File: pkg/knctl/cmd/service/deploy_flags.go [hunk 2]
```diff
@@ -74,12 +79,17 @@ func (s *DeployFlags) Set(cmd *cobra.Command, flagsFactory cmdcore.FlagsFactory)
...
 	cmd.Flags().StringSliceVar(&s.EnvSecrets, "env-secret", nil, "Set environment variable from a secret (format: ENV_KEY=secret-name/key) (can be specified multiple times)")
 	cmd.Flags().StringSliceVar(&s.EnvConfigMaps, "env-config-map", nil, "Set environment variable from a config map (format: ENV_KEY=config-map-name/key) (can be specified multiple times)")
+	cmd.Flags().StringSliceVar(&s.EnvFromConfigMaps, "envfrom-config-map", nil, "Set environment variables as all key-value in a config map (format: config-map-name) (can be specified multiple times)")
 
 	cmd.Flags().Var(newDefaultlessIntValue(&s.ContainerConcurrency), "container-concurrency", "Set container concurrency")
 	cmd.Flags().Var(newDefaultlessIntValue(&s.MinScale), "min-scale", "Set autoscaling rule for minimum number of containers")
 	cmd.Flags().Var(newDefaultlessIntValue(&s.MaxScale), "max-scale", "Set autoscaling rule for maximum number of containers")
+	cmd.Flags().StringSliceVar(&s.VolumeMountSecrets, "mount-secret", nil, "Mount a secret as a volume (format: secret-name=/mount/path) (can be specified multiple times)")
```
Comment (cppforlife) [inline pkg/knctl/cmd/service/deploy_flags.go]: VolumeMountSecrets  -> SecretVolumeMounts
mount-secret -> secret-mount

File: pkg/knctl/cmd/service/deploy_flags.go [hunk 3]
```diff
@@ -74,12 +79,17 @@ func (s *DeployFlags) Set(cmd *cobra.Command, flagsFactory cmdcore.FlagsFactory)
...
+	cmd.Flags().StringSliceVar(&s.VolumeMountConfigMaps, "mount-config-map", nil, "Mount a config map as a volume (format: configmap-name=/mount/path) (can be specified multiple times)")
```
Comment (cppforlife) [inline pkg/knctl/cmd/service/deploy_flags.go]: VolumeMountConfigMaps -> ConfigMapVolumeMounts
mount-config-map -> config-map-mount

File: pkg/knctl/cmd/service/deploy_flags.go [hunk 4]
```diff
@@ -74,12 +79,17 @@ func (s *DeployFlags) Set(cmd *cobra.Command, flagsFactory cmdcore.FlagsFactory)
 	cmd.Flags().StringArrayVarP(&s.EnvVars, "env", "e", nil, "Set environment variable (format: ENV_KEY=value) (can be specified multiple times)")
 	cmd.Flags().StringSliceVar(&s.EnvSecrets, "env-secret", nil, "Set environment variable from a secret (format: ENV_KEY=secret-name/key) (can be specified multiple times)")
 	cmd.Flags().StringSliceVar(&s.EnvConfigMaps, "env-config-map", nil, "Set environment variable from a config map (format: ENV_KEY=config-map-name/key) (can be specified multiple times)")
+	cmd.Flags().StringSliceVar(&s.EnvFromConfigMaps, "envfrom-config-map", nil, "Set environment variables as all key-value in a config map (format: config-map-name) (can be specified multiple times)")
```
Comment (cppforlife) [inline pkg/knctl/cmd/service/deploy_flags.go]: EnvFromConfigMaps -> EnvAllFromConfigMaps
envfrom-config-map -> env-all-from-config-map

**Comment by cppforlife** [comment]
@igsong thanks for contributing, i've left a few comments to fix up before i'll merge this in.

**Comment by igsong** [comment]
@cppforlife I've changed all the things as your guideline. Thank you for reviewing!!

**Comment by cppforlife** [comment]
ill try to merge it on monday. have a good weekend!

**Comment by igsong** [comment]
OK. Have a great weekend!

Thanks,
Ingwon Song

**Comment by igsong** [comment]
@cppforlife Could you let me know what's going on? Thanks.

**Comment by cppforlife** [comment]
@igsong oops, i got held up reviewing some other things. going to try to install knative 0.5.0 today and hopefully merge this in.

**Pull request closed without merging**